app/controllers/api/migrations.php

```diff
@@ -0,0 +1,554 @@
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (array $queries, string $search, Response $response, Database $dbForProject) {
+
+        $queries = Query::parseQueries($queries);
+
+        if (!empty($search)) {
+            $queries[] = Query::search('search', $search);
+        }
+
+        // Get cursor document if there was a cursor query
+        $cursor = Query::getByType($queries, Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE);
+        $cursor = reset($cursor);
+        if ($cursor) {
+            /** @var Query $cursor */
+            $migrationId = $cursor->getValue();
+            $cursorDocument = $dbForProject->getDocument('migrations', $migrationId);
+
+            if ($cursorDocument->isEmpty()) {
+                throw new Exception(Exception::GENERAL_CURSOR_NOT_FOUND, "Migration '{$migrationId}' for the 'cursor' value not found.");
+            }
+
+            $cursor->setValue($cursorDocument);
+        }
+
+        $filterQueries = Query::groupByType($queries)['filters'];
+
+        $response->dynamic(new Document([
+            'migrations' => $dbForProject->find('migrations', $queries),
+            'total' => $dbForProject->count('migrations', $filterQueries, APP_LIMIT_COUNT),
+        ]), Response::MODEL_MIGRATION_LIST);
+    });
+
+App::get('/v1/migrations/:migrationId')
+    ->groups(['api', 'migrations'])
+    ->desc('Get Migration')
+    ->label('scope', 'migrations.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'get')
+    ->label('sdk.description', '/docs/references/migrations/get-migration.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION)
+    ->param('migrationId', '', new UID(), 'Migration unique ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $migrationId, Response $response, Database $dbForProject) {
+        $migration = $dbForProject->getDocument('migrations', $migrationId);
+
+        if ($migration->isEmpty()) {
+            throw new Exception(Exception::MIGRATION_NOT_FOUND, 'Migration not found', 404);
+        }
+
+        $response->dynamic($migration, Response::MODEL_MIGRATION);
+    });
+
+App::post('/v1/migrations/:migrationId')
+    ->groups(['api', 'migrations'])
+    ->desc('Retry Migration')
```

Comment:
POST endpoints are usually to create migrations

Reply:
Hm, yes I agree. CREATE may be a better HTTP call to use. Will update accordingly